### PR TITLE
Copy button

### LINKED ISSUE #73: Add button to copy code to clipboard
A nice little "copy" icon would appear next to the verification code that would automatically put it in the clipboard and show a tooltip after it's clicked saying "Verification code copied to clipboard!" or something to that effect after being clicked and the code copied.

### PR DESCRIPTION
**Pull request by ahmetlii**
I did a button for copying to clipboard. Closes #73

### CODE REVIEW

File: src/SpecialRequestAccount.php [hunk 1]
```diff
@@ -211,7 +211,17 @@ function usernameAndVerificationArea(&$session, $request) {
 			['class' => 'mw-scratch-confirmaccount-verifcode'],
 			ScratchVerification::sessionVerificationCode($session)
 		);
-
+		$form .= Html::element(
+			'button',
+			[
+				'class' => 'mw-scratch-confirmaccount-clickCopy',
+				'onclick' => 'copyToClipboard(document.getElementById("mw-scratch-confirmaccount-verifcode").textContent)',
```
Comment (apple502j) [inline src/SpecialRequestAccount.php:218]: Don't use onclick. Add to the JS file.

File: src/SpecialRequestAccount.php [hunk 2]
```diff
@@ -211,7 +211,17 @@ function usernameAndVerificationArea(&$session, $request) {
...
+				'id' => 'mw-scratch-confirmaccount-clickCopy'
+				'type' => 'button'
+				'value' => 'document.getElementById("mw-scratch-confirmaccount-verifcode").textContent'
```
Comment (apple502j) [inline src/SpecialRequestAccount.php:221]: This is not how things work.

File: src/SpecialRequestAccount.php [hunk 3]
```diff
@@ -211,7 +211,17 @@ function usernameAndVerificationArea(&$session, $request) {
...
+			],
+			'Copy to clipboard'
```
Comment (apple502j) [inline src/SpecialRequestAccount.php:223]: i18n

File: resources/main.js [hunk 4]
```diff
@@ -18,4 +18,20 @@ $(function () {
             textInput.value = value;
         }
     });
-});
\ No newline at end of file
+
+$(function () {
+  Array.prototype.forEach.call(document.getElementsByClassName('mw-scratch-confirmaccount-clickCopy'), el => {
+    el.onclick = function copyToClipboard(temptext) {
+        var tempItem = document.createElement('textarea');
```
Comment (apple502j) [inline resources/main.js:25]: I think reusing existing textarea is enough.

**Review by apple502j — CHANGES_REQUESTED** [review]

File: src/SpecialRequestAccount.php (hunk 1 above)
Comment (ahmetlii) [reply src/SpecialRequestAccount.php:218]: ok

**Comment by ahmetlii** [comment]
@apple502j can you re-review it and merge again?

**Comment by ahmetlii** [comment]
@jacob-g can you just review and accept if possible?

File: resources/main.js [hunk 5]
```diff
@@ -16,6 +17,25 @@ $(function () {
...
+$(function () { document.getElementById("mw-scratch-confirmaccount-click-copy").onclick = function() {
+  copyToClipboard(document.getElementById("mw-scratch-confirmaccount-verifcode"));
+  }
+});
+
+function copyToClipboard(temptext) {
+        var tempItem = document.createElement('textarea');
+        tempItem.value = temptext.innerText;
+        tempItem.style.top = "0";
+        tempItem.style.left = "0";
```
Comment (Kenny2github) [inline resources/main.js:33]: Why have any chance of showing it? -999 and -999 should be fine.

File: resources/main.js [hunk 6]
```diff
@@ -16,6 +17,25 @@ $(function () {
...
+        tempItem.style.position = "fixed";
+        document.body.appendChild(tempItem);
+        tempItem.focus();
+        tempItem.select();
+        document.execCommand('copy');
+        tempItem.parentElement.removeChild(tempItem);
+        alert(mw.message( 'scratch-confirmaccount-click-copy-alert').plain());
```
Comment (Kenny2github) [inline resources/main.js:40]: Please use a [bubble notification](https://www.mediawiki.org/wiki/Bubble_notifications) for this.

File: resources/main.js (hunk 5 above)
Comment (Kenny2github) [inline resources/main.js:33]: Why have any chance of showing it?
```suggestion
        tempItem.style.top = "-999px";
        tempItem.style.left = "-999px";
```

File: src/SpecialRequestAccount.php [hunk 7]
```diff
@@ -208,10 +208,20 @@ function usernameAndVerificationArea(&$session, $request) {
 		);
 		$form .= Html::element(
 			'p',
-			['class' => 'mw-scratch-confirmaccount-verifcode'],
+			['class' => 'mw-scratch-confirmaccount-verifcode',
+		   'id' => 'mw-scratch-confirmaccount-verifcode'
```
Comment (Kenny2github) [inline src/SpecialRequestAccount.php:212]: Weird indentation here

**Review by Kenny2github — CHANGES_REQUESTED** [review]

File: extension.json [hunk 8]
```diff
@@ -35,7 +35,19 @@
...
+			],
+			"dependencies": [
+			 "mediawiki.util",
+			 "mediawiki.api",
+			 "mediawiki.confirmCloseWindow",
+			 "jquery.textSelection",
+			 "jquery.spinner",
+			 "oojs-ui-core",
+			 "oojs-ui-widgets"
+			]
```
Comment (Kenny2github) [inline extension.json:50]: Weird indentation here

**Review by Kenny2github — CHANGES_REQUESTED** [review]
See also the conversation I unresolved

**Comment by SausageMcSauce** [comment]
This might not be considered a problem (and I am not going against anyone), but this pull request contains 57 commits from the requestor, and this getting merged might litter the commit history of this repository with all of those commits, some of which contain very minor changes.

**Comment by Kenny2github** [comment]
@SausageMcSauce Squash merging is a thing. All commits from this branch will be combined into one commit when merging.

**Review by Kenny2github — APPROVED** [review]

**Pull request merged**